scrapd/core/formatter.py

```diff
@@ -96,8 +107,20 @@ class JSONFormatter(Formatter):
 
     __format_name__ = 'json'
 
+    def to_json_string(self, results):
+        """
+        Convert dict of parsed fields to JSON string.
+
+        :param results dict: results of scraping APD news site
+
+        :rtype: str
+        """
+
+        return json.dumps(results, sort_keys=True, indent=2, default=self.date_serialize)
+
     def printer(self, results, **kwargs):  # noqa: D102
-        print(json.dumps(results, sort_keys=True, indent=2), file=self.output)
+        json_string = self.to_json_string(results, **kwargs)
+        print(json_string, file=self.output)
```

Comment:
You don't need to create an extra function here. `to_json_string()` is not really necessary, you can simply put:

```python
print(json.dumps(results, sort_keys=True, indent=2, default=self.date_serialize), file=self.output)
```

Reply:
I wanted that line to be in a method with a return value for easier testing.